Unity/Champloo/Assets/Scripts/Player/Movement States/OnMovementSpecial.cs

```diff
@@ -50,6 +50,11 @@ protected virtual IEnumerator TimingCoroutine()
         OnEnd();
         yield return new WaitForSeconds(cooldownTime);
         OnCooledDown();
+        if (rechargeTime > 0)
```

Comment:
Honestly we should do this for all of the yeild statements in this function and the other timing functions. Could you do a separate PR for those?